### PR TITLE
Openshift SCC object to mount nfs volumes

### PR DESCRIPTION
**Pull request by haasad**
This PR adds a security context constraint object for openshift distributions, which allows mounting of nfs volumes.

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [ ] Update the documentation.
- [x] Update the ./CHANGELOG.md.

### CODE REVIEW

File: component/openshiftscc.jsonnet [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+local isOpenshift = std.startsWith(inv.parameters.facts.distribution, 'openshift');
+local instance = inv.parameters._instance;
+
+local nfsMountScc = {
+  apiVersion: 'security.openshift.io/v1',
+  kind: 'SecurityContextConstraints',
+  metadata: {
+    name: 'hostmount-anyuid-' + instance,
+    namespace: params.namespace,
```
Comment (simu) [inline component/openshiftscc.jsonnet:16]: I'd suggest adding the recommended labels according to https://syn.tools/syn/references/labels-guide.html so it's explicitly clear that the SCC is Syn-managed and which component is responsible for it.

**Review by simu — APPROVED** [review]
LGTM overall

File: component/openshiftscc.jsonnet (hunk 1 above)
Comment (haasad) [reply component/openshiftscc.jsonnet:16]: I added:
```
'app.kubernetes.io/name': 'hostmount-anyuid-' + instance,                                                                                                           
'app.kubernetes.io/managed-by': 'syn',                                                                                                                              
```
Should I also add sth like:
```
app.kubernetes.io/part-of: syn-component-nfs-subdir-external-provisioner
```
Also I'm unsure what 'app.kubernetes.io/instance' should refer to in this context.

Comment (simu) [reply component/openshiftscc.jsonnet:16]: I think just `app.kubernetes.io/managed-by: syn` and `app.kubernetes.io/part-of: syn-component-nfs-subdir-external-provisioner` would be sufficient.

Alternatively, you could have `app.kubernetes.io/instance` refer to the nfs-subdir-external-provisioner instance while `app.kubernetes.io/component` could be set to `nfs-subdir-external-provisioner`. In that case, you could omit the `part-of` label.

Comment (haasad) [reply component/openshiftscc.jsonnet:16]: Thanks, I went with the "alternative" approach to make it as clear as possible.

File: component/openshiftscc.jsonnet [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    labels: {
+      'app.kubernetes.io/name': 'hostmount-anyuid-' + instance,
+      'app.kubernetes.io/instance': 'hostmount-anyuid-' + instance,
```
Comment (simu) [inline component/openshiftscc.jsonnet:19]:
```suggestion
      'app.kubernetes.io/instance': instance,
```

File: component/openshiftscc.jsonnet (hunk 1 above)
Comment (simu) [reply component/openshiftscc.jsonnet:16]: See my suggestion, I would just do `'app.kubernetes.io/instance': instance`

**Review by simu — APPROVED** [review]

**Pull request merged**